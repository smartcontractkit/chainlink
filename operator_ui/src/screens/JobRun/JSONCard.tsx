import React from 'react'

import Card from '@material-ui/core/Card'
import CardContent from '@material-ui/core/CardContent'

import { PrettyJson } from 'src/components/Syntax/PrettyJson'

interface Props {
  run: JobRunPayload_Fields
}

export const JSONCard: React.FC<Props> = ({ run }) => {
  // The run inputs are returned as a string instead of JSON, but we want to
  // display it as JSON.
  const obj = React.useMemo(() => {
    const { inputs, outputs, taskRuns, ...rest } = run

    let inputsObj = {}
    try {
      inputsObj = JSON.parse(inputs)
    } catch (e) {
      inputsObj = {}
    }

    return {
      ...rest,
      inputs: inputsObj,
      outputs,
      taskRuns,
    }
  }, [run])

  return (
    <Card>
      <CardContent>
        <PrettyJson object={obj} />
      </CardContent>
    </Card>
  )
}
