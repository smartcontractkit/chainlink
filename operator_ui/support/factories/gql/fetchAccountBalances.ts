// buildETHKey builds a eth key for the FetchETHKeys query.
export function buildETHKey(
  overrides?: Partial<AccountBalancesPayload_ResultsFields>,
): AccountBalancesPayload_ResultsFields {
  return {
    __typename: 'EthKey',
    address: '0x0000000000000000000000000000000000000001',
    chain: {
      __typename: 'Chain',
      id: '42',
    },
    ethBalance: '0.100000000000000000',
    isFunding: false,
    linkBalance: '1000000000000000000',
    ...overrides,
  }
}

// buildETHKeys builds a list of eth keys.
export function buildETHKeys(): ReadonlyArray<AccountBalancesPayload_ResultsFields> {
  return [
    buildETHKey(),
    buildETHKey({
      address: '0x0000000000000000000000000000000000000002',
    }),
  ]
}
