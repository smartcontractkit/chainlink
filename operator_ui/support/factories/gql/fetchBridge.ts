// buildBridgePayloadFields builds the bridge fields.
export function buildBridgePayloadFields(
  overrides?: Partial<BridgePayload_Fields>,
): BridgePayload_Fields {
  return {
    __typename: 'Bridge',
    id: 'bridge-api',
    name: 'bridge-api',
    url: 'http://bridge.com',
    confirmations: 1,
    minimumContractPayment: '0',
    outgoingToken: 'outgoing1',
    ...overrides,
  }
}
