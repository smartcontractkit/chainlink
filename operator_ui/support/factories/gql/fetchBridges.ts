// buildBridge builds a Bridge for the FetchBridges query.
export function buildBridge(
  overrides?: Partial<BridgesPayload_ResultsFields>,
): BridgesPayload_ResultsFields {
  return {
    __typename: 'Bridge',
    id: 'bridge-api',
    name: 'bridge-api',
    url: 'http://bridge.com',
    confirmations: 1,
    minimumContractPayment: '0',
    ...overrides,
  }
}

// buildsBridges builds a list of bridges.
export function buildBridges(): ReadonlyArray<BridgesPayload_ResultsFields> {
  return [
    buildBridge({
      id: 'bridge-api1',
      name: 'bridge-api1',
      url: 'http://bridge1.com',
      confirmations: 1,
      minimumContractPayment: '100',
    }),
    buildBridge({
      id: 'bridge-api2',
      name: 'bridge-api2',
      url: 'http://bridge2.com',
      confirmations: 2,
      minimumContractPayment: '200',
    }),
  ]
}
