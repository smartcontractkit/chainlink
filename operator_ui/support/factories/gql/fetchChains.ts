import isoDate, { MINUTE_MS } from 'test-helpers/isoDate'

// buildChains builds a chain for the FetchChains query.
export function buildChain(
  overrides?: Partial<ChainsPayload_ResultsFields>,
): ChainsPayload_ResultsFields {
  const minuteAgo = isoDate(Date.now() - MINUTE_MS)

  return {
    __typename: 'Chain',
    id: '5',
    enabled: true,
    createdAt: minuteAgo,
    ...overrides,
  }
}

// buildsChains builds a list of chains.
export function buildChains(): ReadonlyArray<ChainsPayload_ResultsFields> {
  const minuteAgo = isoDate(Date.now() - MINUTE_MS)

  return [
    buildChain({
      id: '5',
      enabled: true,
      createdAt: minuteAgo,
    }),
    buildChain({
      id: '42',
      enabled: true,
      createdAt: minuteAgo,
    }),
  ]
}
