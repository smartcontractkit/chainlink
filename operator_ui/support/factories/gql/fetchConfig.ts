// buildConfigItem builds the config item fields.
export function buildConfigItem(
  overrides?: Partial<Config_ItemsFields>,
): Config_ItemsFields {
  return {
    __typename: 'ConfigItem',
    key: 'BAND',
    value: 'Major Lazer',
    ...overrides,
  }
}
