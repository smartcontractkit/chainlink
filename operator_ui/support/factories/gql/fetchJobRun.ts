import isoDate, { MINUTE_MS } from 'test-helpers/isoDate'

export function buildRun(
  overrides?: Partial<JobRunPayload_Fields>,
): JobRunPayload_Fields {
  const minuteAgo = isoDate(Date.now() - MINUTE_MS)
  const twoMinutesAgo = isoDate(Date.now() - MINUTE_MS * 2)

  return {
    __typename: 'JobRun',
    id: '1',
    allErrors: [],
    createdAt: twoMinutesAgo,
    fatalErrors: [],
    finishedAt: minuteAgo,
    inputs: '',
    job: {
      id: '10',
      name: 'job 1',
      observationSource: '',
    },
    outputs: [],
    status: 'COMPLETED',
    taskRuns: [],
    ...overrides,
  }
}

export function buildTaskRun(
  overrides?: Partial<JobRunPayload_TaskRunsFields>,
): JobRunPayload_TaskRunsFields {
  const minuteAgo = isoDate(Date.now() - MINUTE_MS)
  const twoMinutesAgo = isoDate(Date.now() - MINUTE_MS * 2)

  return {
    __typename: 'TaskRun',
    id: '00000000-0000-0000-0000-000000000001',
    createdAt: twoMinutesAgo,
    dotID: 'parse_request',
    error: 'data: parameter is empty',
    finishedAt: minuteAgo,
    output: 'null',
    type: 'jsonparse',
    ...overrides,
  }
}
