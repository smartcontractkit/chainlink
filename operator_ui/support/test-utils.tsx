import React from 'react'
import { Provider as ReduxProvider } from 'react-redux'
import { MemoryRouter } from 'react-router-dom'
import { render, RenderOptions, RenderResult } from '@testing-library/react'

import { MuiThemeProvider } from '@material-ui/core/styles'

import createStore from 'src/createStore'
import { theme } from 'src/theme'

const AllTheProviders: React.FC = ({ children }) => {
  return (
    <MuiThemeProvider theme={theme}>
      <ReduxProvider store={createStore()}>{children}</ReduxProvider>
    </MuiThemeProvider>
  )
}

const customRender = (
  ui: React.ReactElement,
  options?: Omit<RenderOptions, 'wrapper'>,
): RenderResult =>
  render(ui, {
    wrapper: AllTheProviders,
    ...options,
  })
interface RenderWithRouterProps {
  initialEntries?: string[]
}

// renderWithRouter behaves like 'render' except it wraps the provided ui in a
// Router.
//
// Use this when you need a router in your tests for page components and any
// component which uses react-router hooks.
const renderWithRouter = (
  ui: React.ReactElement,
  { initialEntries }: RenderWithRouterProps = { initialEntries: ['/'] },
  options?: Omit<RenderOptions, 'wrapper'>,
): RenderResult => {
  return {
    ...customRender(
      <MemoryRouter initialEntries={initialEntries}>{ui}</MemoryRouter>,
      options,
    ),
  }
}

export * from '@testing-library/react'
export { customRender as render, renderWithRouter }
