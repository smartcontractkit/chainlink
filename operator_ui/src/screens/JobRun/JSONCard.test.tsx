import * as React from 'react'
import { buildRun } from 'support/factories/gql/fetchJobRun'

import { render, screen } from 'support/test-utils'

import { JSONCard } from './JSONCard'

const { getByTestId } = screen

describe('JSONCard', () => {
  it('renders the run as json', () => {
    const start = '2020-01-03T22:45:00.166261Z'
    const end1m = '2020-01-03T22:46:00.166261Z'

    const run = buildRun({
      createdAt: start,
      finishedAt: end1m,
    })

    render(<JSONCard run={run} />)

    expect(getByTestId('pretty-json').textContent).toMatchSnapshot()
  })
})
