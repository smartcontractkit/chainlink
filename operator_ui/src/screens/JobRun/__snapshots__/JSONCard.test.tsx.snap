// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`JSONCard renders the run as json 1`] = `"{  \\"__typename\\": \\"JobRun\\",  \\"id\\": \\"1\\",  \\"allErrors\\": [  ],  \\"createdAt\\": \\"2020-01-03T22:45:00.166261Z\\",  \\"fatalErrors\\": [  ],  \\"finishedAt\\": \\"2020-01-03T22:46:00.166261Z\\",  \\"job\\": {    \\"id\\": \\"10\\",    \\"name\\": \\"job 1\\",    \\"observationSource\\": \\"\\"  },  \\"status\\": \\"COMPLETED\\",  \\"inputs\\": {  },  \\"outputs\\": [  ],  \\"taskRuns\\": [  ]}"`;
